File: clients/cpp/examples/publish.cpp
/**
 * @file publish.cpp
 * @brief Example: Publishing messages with NexusD C++ client
 */

#include <nexusd_client/client.hpp>
#include <iostream>
#include <thread>
#include <chrono>

int main() {
    try {
        // Connect to daemon
        nexusd::Client client("localhost:5672");
        std::cout << "Connected to NexusD\n";
        
        // Publish a simple message
        auto result = client.publish(
            "sensors/temperature",
            R"({"value": 23.5, "unit": "celsius"})",
            {.content_type = "application/json"}
        );
        std::cout << "Published: message_id=" << result.message_id
                  << ", subscribers=" << result.subscriber_count << "\n";
        
        // Publish with retain
        result = client.publish(
            "config/settings",
            R"({"theme": "dark", "language": "en"})",
            {.content_type = "application/json", .retain = true}
        );
        std::cout << "Published retained: message_id=" << result.message_id << "\n";
        
        // Publish multiple messages
        for (int i = 0; i < 5; ++i) {
            std::string payload = R"({"value": )" + std::to_string(45 + i) + R"(, "unit": "%"})";
            result = client.publish(
                "sensors/humidity",
                payload,
                {.content_type = "application/json"}
            );
            std::cout << "Published humidity #" << i
                      << ": subscribers=" << result.subscriber_count << "\n";
            std::this_thread::sleep_for(std::chrono::milliseconds(500));
        }
        
        std::cout << "Done!\n";
        return 0;
        
    } catch (const std::exception& e) {
        std::cerr << "Error: " << e.what() << "\n";
        return 1;
    }
}

File: clients/node/examples/publish.ts
/**
 * Example: Publishing messages with NexusD Node.js client
 */

import { NexusdClient } from '../src';

async function main() {
    // Connect to daemon
    const client = new NexusdClient('localhost:5672');
    console.log('Connected to NexusD\n');

    try {
        // Publish a simple message
        let result = await client.publish(
            'sensors/temperature',
            Buffer.from(JSON.stringify({ value: 23.5, unit: 'celsius' })),
            { contentType: 'application/json' }
        );
        console.log(`Published: message_id=${result.messageId}, subscribers=${result.subscriberCount}`);

        // Publish with retain
        result = await client.publish(
            'config/settings',
            Buffer.from(JSON.stringify({ theme: 'dark', language: 'en' })),
            { contentType: 'application/json', retain: true }
        );
        console.log(`Published retained: message_id=${result.messageId}`);

        // Publish multiple messages
        for (let i = 0; i < 5; i++) {
            result = await client.publish(
                'sensors/humidity',
                Buffer.from(JSON.stringify({ value: 45 + i, unit: '%' })),
                { contentType: 'application/json' }
            );
            console.log(`Published humidity #${i}: subscribers=${result.subscriberCount}`);
            await new Promise((resolve) => setTimeout(resolve, 500));
        }

        console.log('\nDone!');
    } finally {
        client.close();
    }
}

main().catch(console.error);
